app/app/utils.py

```diff
@@ -185,7 +185,20 @@ def get_upload_filename(instance, filename):
     return f"docs/{getattr(instance, '_path', '')}/{salt}/{file_path}"
 
 
-def sync_profile(handle, user=None, hide_profile=True):
+def sync_profile(handle, user=None, hide_profile=True, delay_okay=False):
+    from dashboard.models import Profile
+    handle = handle.strip().replace('@', '').lower()
+    profile = Profile.objects.filter(handle=handle).exists()
+    delay = delay_okay and profile
+    if delay:
+        from dashboard.tasks import sync_profile as sync_profile_task
+        user_pk = user.pk if user else None
+        sync_profile_task.delay(handle, user_pk, hide_profile)
+    else:
+        actually_sync_profile(handle, user=user, hide_profile=hide_profile)
+
+
+def actually_sync_profile(handle, user=None, hide_profile=True):
```

Comment:
many of the chat related tasks here could be done on a delay as well; a further PR  would possibly do that so that upon first login were not blocking on things like chat tasks to create the profile